build.sbt

```diff
@@ -614,18 +618,47 @@ lazy val scalalib =
         for {
           srcDir <- sourceDirectories
           normSrcDir = normPath(srcDir)
-          src <- (srcDir ** "*.scala").get
+          scalaGlob  = srcDir.toGlob / ** / "*.scala"
+          patchGlob  = srcDir.toGlob / ** / "*.scala.patch"
+
+          (sourcePath, _) <- fileTreeView.value.list(Seq(scalaGlob, patchGlob))
+          path = normPath(sourcePath.toFile).substring(normSrcDir.length)
         } {
-          val normSrc = normPath(src)
-          val path    = normSrc.substring(normSrcDir.length)
-          val useless =
-            path.contains("/scala/collection/parallel/") ||
-              path.contains("/scala/util/parsing/")
-          if (!useless) {
+          def addSource(path: String, source: File): Unit = {
             if (paths.add(path))
-              sources += src
+              sources += source
             else
-              s.log.debug(s"not including $src")
+              s.log.debug(s"not including $path")
+          }
+
+          if (patchGlob.matches(sourcePath)) {
+            val sourceName      = path.stripSuffix(".patch")
+            val scalaSourcePath = scalaSrcDir / sourceName
+            val outputFile      = scalaSrcDir / (sourceName + ".patched")
```

Comment:
The output file should be somewhere in a `target/` directory, since it will contain generated files. Additionally, that way it can have the extension `.scala` instead of `.patched`.